runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -90,39 +142,104 @@ AOTCacheClassRecord::create(uintptr_t id, const AOTCacheClassLoaderRecord *class
    return new (ptr) AOTCacheClassRecord(id, classLoaderRecord, hash, romClass);
    }
 
+AOTCacheClassRecord *
+AOTCacheClassRecord::create(size_t nameLength, const AOTCacheClassLoaderRecord *classLoaderRecord)
+   {
+   void *ptr = AOTCacheRecord::allocate(nameLength);
+   return new (ptr) AOTCacheClassRecord(classLoaderRecord);
+   }
+
 void
 AOTCacheClassRecord::subRecordsDo(const std::function<void(const AOTCacheRecord *)> &f) const
    {
    f(_classLoaderRecord);
    }
 
+AOTCacheClassRecord *
+AOTCacheClassRecord::read(FILE *f, const std::vector<AOTCacheClassLoaderRecord *> &classLoaderRecords)
```

Comment:
You can unify the records' `read()` API by passing the whole snapshot loading context to it (which would include all the id-record vectors).